src/frontend/src/flows/register.ts

```diff
@@ -59,18 +60,20 @@ const init = (): Promise<LoginResult | null> =>
       try {
         const pendingIdentity = WebAuthnIdentity.create({
           publicKey: creationOptions(),
-        }).catch((error) => {
-          resolve(apiResultToLoginResult({ kind: "authFail", error }));
-          // We can never get here, but TS doesn't understand that
-          return 0 as unknown as WebAuthnIdentity;
         });
         await tick();
         // Kick-start both the captcha creation and the identity
-        const captcha = makeCaptcha();
-        const identity = await pendingIdentity;
-        await captcha;
-        const result = await confirmRegister(captcha, identity, alias);
-        resolve(result);
+        Promise.all([makeCaptcha(), pendingIdentity])
+          .catch((error) => {
+            resolve(apiResultToLoginResult({ kind: "authFail", error }));
+            // We can never get here, but TS doesn't understand that
+            return 0 as unknown as [Challenge, WebAuthnIdentity];
+          })
+          .then(([captcha, identity]) => {
+            confirmRegister(Promise.resolve(captcha), identity, alias).then(
+              resolve
+            );
+          });
```

Comment:
Does this do what you want?
- It skips the confirmRegister if any of makeCaptcha or pendingIdentity fail.
- The entire chain is infallible, due to the last line.
- No resolve.
```
        Promise.all([makeCaptcha(), pendingIdentity])
          .catch(error => throw apiResultToLoginResult({ kind: "authFail", error }))
          .then(([captcha, identity]) => confirmRegister(captcha, identity, alias))
          .catch(e => e);
```
Note that any failures in the confirmRegister might not be LoginResult errors unless you map those as well.  It looks as if there are several different error types that can happen there.  I feel tempted to think that if re-rendering fails something is really messed up and maybe the right thing to do is to call reload(), effectively making the confirmRegister infallible.  It will never report anything but success.  Alternatively, you could map all errors to login result errors with:
```
        Promise.all([makeCaptcha(), pendingIdentity])
          .then(([captcha, identity]) => confirmRegister(captcha, identity, alias))
          .catch(error => apiResultToLoginResult({ kind: "authFail", error }));
```

Reply:
Does this do what you want?

It skips the confirmRegister if any of makeCaptcha or pendingIdentity fail.
The entire chain is infallible, due to the last line.
No resolve.

---

Yes, this is what I want; I believe this is what the original author meant to do.


> Alternatively, you could map all errors to login result errors with:

I like this. Let me edit.